node/packages/parser/src/ethereum/index.ts

```diff
@@ -171,6 +175,19 @@ export class TransactionParser {
       }
     })
 
+    const metaData: TxMetadata = (() => {
```

Comment:
I was initially thinking the contract parsers would return possible tx metadata depending on what it saw via contract interaction. If we went this route and made use of the input data to determine interaction, we would be able to uncouple the metadata from the parsed transfers. Not only that, but it seems like we would need the flexibility to do more custom metadata parsing that would have to belong in the contract parsers regardless (though maybe this is what you are implying with the follow up PR).

Reply:
I am also not a fan of this anon function. It is just a more verbose creation of an object versus:
```
const metaData: TxMetaData = {
  buyTx: getBuyTx(pTx),
  sellTx: getSellTx(pTx)
}
```

Reply:
Simplified in https://github.com/shapeshift/unchained/pull/247/commits/f033e0557e89e877934340067a572504c431f00f

I believe we are aligned re: the contract interaction determining the metadata. What I wanted to do first was just copy all the logic across from `web` in a naive way, and then get `web` using the transaction `data` it received from `unchained`.

Mitigates risk, and lets us firm up tests against existing logic (albeit in a new repo) before making bigger refactors.